cmd/bosun/expr/funcs.go

```
@@ -595,6 +603,56 @@ func Merge(e *State, T miniprofiler.Timer, series ...*Results) (*Results, error)
 	return res, nil
 }
 
+func LeftJoin(e *State, T miniprofiler.Timer, keysCSV, columnsCSV string, rowData ...*Results) (*Results, error) {
+	res := &Results{}
+	dataWidth := len(rowData)
+	if dataWidth == 0 {
+		return res, fmt.Errorf("leftjoin requires at least one item to populate rows")
+	}
+	keyColumns := strings.Split(keysCSV, ",")
+	dataColumns := strings.Split(columnsCSV, ",")
+	keyWidth := len(keyColumns)
+	keyIndex := make(map[string]int)
+	for i, v := range keyColumns {
+		keyIndex[v] = i
+	}
+	if len(dataColumns) != dataWidth {
```

can do this check even sooner. fail fast.